File: airbyte-connector-builder-server/src/test/java/io/airbyte/connector_builder/command_runner/MockProcess.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connector_builder.command_runner;

import java.io.InputStream;
import java.io.OutputStream;

/**
 * Mocks the Process class.
 */
public class MockProcess extends Process {

  int exitValue;
  InputStream inputStream;
  InputStream errorStream;
  OutputStream outputStream;

  public MockProcess(
                     final int exitValue,
                     final InputStream errorStream,
                     final InputStream inputStream,
                     final OutputStream outputStream) {
    this.exitValue = exitValue;
    this.errorStream = errorStream;
    this.inputStream = inputStream;
    this.outputStream = outputStream;
  }

  @Override
  public InputStream getErrorStream() {
    return this.errorStream;
  }

  @Override
  public InputStream getInputStream() {
    return this.inputStream;
  }

  @Override
  public OutputStream getOutputStream() {
    return this.outputStream;
  }

  @Override
  public int exitValue() {
    return this.exitValue;
  }

  @Override
  public int waitFor() {
    return 0;
  }

  @Override
  public void destroy() {}

  @Override
  public ProcessHandle.Info info() {
    return null;
  }

  @Override
  public boolean isAlive() {
    return false;
  }

}

File: airbyte-connector-builder-server/src/test/kotlin/io/airbyte/connectorbuilder/commandrunner/MockProcess.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.commandrunner

import java.io.InputStream
import java.io.OutputStream

/**
 * Mocks the Process class.
 */
class MockProcess(
  private var exitValue: Int,
  private var errorStream: InputStream,
  private var inputStream: InputStream,
  private var outputStream: OutputStream,
) : Process() {
  override fun getErrorStream(): InputStream = this.errorStream

  override fun getInputStream(): InputStream = this.inputStream

  override fun getOutputStream(): OutputStream = this.outputStream

  override fun exitValue(): Int = this.exitValue

  override fun waitFor(): Int = 0

  override fun destroy() {}

  override fun info(): ProcessHandle.Info? = null

  override fun isAlive(): Boolean = false
}
